SrcMLCppAPI/SrcMLCppAPI.cpp

```diff
@@ -3,10 +3,45 @@
 #include "stdafx.h"
 #include "srcml.h"
 #include "SrcMLCppAPI.h"
-
+#include <iostream>
+#include <string>
 using namespace System;
 using namespace System::Runtime::InteropServices;
+void SetMetaData(srcml_unit* unit, SourceData* archiveData){
```

Comment:
Most of the srcml_set_* functions used here aren't specific to the unit. Is this thread safe? If we convert two source files at the same time, each with completely different settings, will they each be done correctly?